src/encounter.py

```diff
@@ -30,19 +36,21 @@ def take_bonus_action(self, action: Action) -> None:
 
 
 class Encounter:
-    def __init__(self):
+    def __init__(self) -> None:
         self.__grid: Grid = Grid(3, 3)
         self.__players: Set[Creature] = set()
         self.__npc: Set[Creature] = set()
         self.__active: Creature | None = None
-        self.__initiative_queue: Queue = Queue()
+        self.__initiative_queue: Queue[Creature] = Queue()
 
     @property
-    def active_creature(self):
+    def active_creature(self) -> Creature:
+        if self.__active is None:
+            raise RuntimeError("No active creature added to encounter!")
```

Comment:
There should always be an active creature after determining initiative (happens in "initialize"). If you think returning None is not descriptive enough, please add a more specific exception type (EncounterNotInitializedException ?)

Reply:
I agree, except that if we really must always have an active creature, we should require that in the Init and not define `__active` as optional.
I can also create a specific exception if you'd prefer that. YDT?

Reply:
The issue with requiring from init is that it would also force us to also take the players as parameter, and I kinda cringe about both adding players in init and as a method - so that would leave us with an immutable encounter, and the DM will be sad :(
Maybe a more specific exception is the way to go here.
Another suggestion is to leave it None and add this for clarification:
`
    @property
    def is_live(self) -> bool:
        return bool(self.__active)
`

Your call. Feel free to resolve this when you're done